integration-tests/bats/merge.bats

```diff
@@ -851,3 +846,23 @@ SQL
     log_status_eq 1
     [[ "$output" =~ "table with same name deleted and modified" ]] || false
 }
+
+@test "merge: dolt merge commits successful non fast forward merge" {
+    dolt branch other
+    dolt sql -q "INSERT INTO test1 VALUES (1,2,3)"
+    dolt commit -am "add (1,2,3) to test1";
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO test1 VALUES (2,3,4)"
+    dolt commit -am "add (2,3,4) to test1";
+
+    dolt checkout main
+    run dolt merge other -m "merge other"
```

Comment:
This would be a good place to test out `--no-commit`, too. You could run the merge first with `--no-commit`, test that the changes are staged, but not committed, then `reset --hard` and run the merge with a commit message.